src/compiler/checker.ts

```diff
@@ -37014,7 +37014,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (!witnesses) {
                 return false;
             }
-            const operandConstraint = getBaseConstraintOrType(checkExpressionCached((node.expression as TypeOfExpression).expression));
+            const operandConstraint = getBaseConstraintOfType(checkExpressionCached((node.expression as TypeOfExpression).expression)) || unknownType;

```

Comment:
This change as is breaks the following case:

```ts
// error on return type because switch is not seen as exhaustive
function woo(x: string): number {
    switch (typeof x) {
        case "string":
            return 1;
    }
}
```
The problem is that `getBaseConstraintOfType` returns `undefined` for primitive types like `string`, in addition to returning `undefined` for an unconstrained type parameter, which is why we use `getBaseConstraintOrType` currently.
I think you might need to explicitly check the unconstrained type parameter case and use `unknown` as the operand constraint then.

Reply:
Good catch 👍 I pushed out a fix for this - could you take a look?